includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1056,30 +1053,70 @@ public function normalize_state() {
 		$shipping_state   = ! empty( $_POST['shipping_state'] ) ? wc_clean( $_POST['shipping_state'] ) : '';
 
 		if ( $billing_state && $billing_country ) {
-			$valid_states = WC()->countries->get_states( $billing_country );
-
-			// Valid states found for country.
-			if ( ! empty( $valid_states ) && is_array( $valid_states ) && sizeof( $valid_states ) > 0 ) {
-				foreach ( $valid_states as $state_abbr => $state ) {
-					if ( preg_match( '/' . preg_quote( $state ) . '/i', $billing_state ) ) {
-						$_POST['billing_state'] = $state_abbr;
-					}
-				}
-			}
+			$_POST['billing_state'] = $this->get_normalized_state( $billing_state, $billing_country );
 		}
 
 		if ( $shipping_state && $shipping_country ) {
-			$valid_states = WC()->countries->get_states( $shipping_country );
+			$_POST['shipping_state'] = $this->get_normalized_state( $shipping_state, $shipping_country );
+		}
+	}
 
-			// Valid states found for country.
-			if ( ! empty( $valid_states ) && is_array( $valid_states ) && sizeof( $valid_states ) > 0 ) {
-				foreach ( $valid_states as $state_abbr => $state ) {
-					if ( preg_match( '/' . preg_quote( $state ) . '/i', $shipping_state ) ) {
-						$_POST['shipping_state'] = $state_abbr;
-					}
+	/**
+	 * Gets the normalized state/county field because in some
+	 * cases, the state/county field is formatted differently from
+	 * what WC is expecting and throws an error. An example
+	 * for Ireland the county dropdown in Chrome shows "Co. Clare" format.
+	 *
+	 * @since 4.9.0
+	 *
+	 * @param string $state   Full state name or an already normalized abbreviation.
+	 * @param string $country Two-letter country code.
+	 *
+	 * @return string Normalized state abbreviation.
+	 */
+	public function get_normalized_state( $state, $country ) {
+		$wc_valid_states = $country ? WC()->countries->get_states( $country ) : [];
+
+		if ( $state && $country && is_array( $wc_valid_states ) && sizeof( $wc_valid_states ) > 0 ) {
+
+			// If it's already normalized, skip.
+			if ( in_array( $state, array_keys( $wc_valid_states ) ) ) {
+				return $state;
+			}
+
+			$match_from_state_input = false;
+
+			// China - Adapt dropdown values from Chrome and accept manually typed values like 云南.
```

Comment:
Is it worth it to add a link to the valid states/regions for CN?

Reply:
https://github.com/woocommerce/woocommerce/blob/master/i18n/states.php

This link? I think it is worth it.